shared/core/types/relay_test.go

```diff
@@ -0,0 +1,116 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRelayValidate(t *testing.T) {
+	testCases := []struct {
+		name     string
+		relay    Relay
+		expected error
+	}{
+		{
+			name: "valid Relay: JSONRPC",
+			relay: Relay{
+				RelayPayload: &Relay_JsonRpcPayload{
+					JsonRpcPayload: &JSONRPCPayload{JsonRpc: "2.0", Method: "eth_blockNumber"},
+				},
+			},
+		},
+		{
+			name: "valid Relay: REST",
+			relay: Relay{
+				RelayPayload: &Relay_RestPayload{
+					RestPayload: &RESTPayload{Contents: `{"field1": "value1", "field2": "value2"}`},
+				},
+			},
+		},
+		{
+			name:     "invalid Relay: missing payload",
+			expected: errInvalidRelayPayload,
+		},
+		{
+			name: "invalid Relay: invalid JSONRPC Payload",
+			relay: Relay{
+				RelayPayload: &Relay_JsonRpcPayload{
+					JsonRpcPayload: &JSONRPCPayload{JsonRpc: "foo"},
+				},
+			},
+			expected: errInvalidJSONRPC,
+		},
+		{
+			name: "invalid Relay: invalid REST Payload",
+			relay: Relay{
+				RelayPayload: &Relay_RestPayload{
+					RestPayload: &RESTPayload{Contents: "foo"},
+				},
+			},
+			expected: errInvalidRESTPayload,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			err := testCase.relay.Validate()
+			require.ErrorIs(t, err, testCase.expected)
+		})
+	}
+}
+
+func TestJsonRpcValidate(t *testing.T) {
+	testCases := []struct {
+		name     string
+		payload  JSONRPCPayload
+		expected error
+	}{
+		{
+			name:    "valid JSONRPC",
+			payload: JSONRPCPayload{JsonRpc: "2.0", Method: "eth_blockNumber"},
+		},
+		{
+			name:     "invalid JSONRPC: invalid JsonRpc field value",
+			payload:  JSONRPCPayload{JsonRpc: "foo", Method: "eth_blockNumber"},
+			expected: errInvalidJSONRPC,
+		},
+		{
+			name:     "invalid JSONRPC: Method field not set",
+			payload:  JSONRPCPayload{JsonRpc: "2.0"},
+			expected: errInvalidJSONRPCMissingMethod,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			err := testCase.payload.Validate()
+			require.ErrorIs(t, err, testCase.expected)
+		})
+	}
+}
+
+func TestRESTValidate(t *testing.T) {
```

Comment:
NIT: Can you update all of these to `TestRelay_ValidateREST`, `TestRelay_ValidJSONRPC`, etc...so we can run them with `go test -run TestRelay`

Reply:
Great suggestion, thank you. Fixed.